base/range.jl

```diff
@@ -773,7 +773,7 @@ let bigints = Union{Int, UInt, Int64, UInt64, Int128, UInt128}
         else
```

Comment:
1. If we decide to always return a `T`, then I suggest to 
```julia
if s isa Unsigned || -1 <= s <= 1 || s == -s
    a = div(diff, s) % T
elseif s < 0
    a = div(unsigned(-diff), -s) % T
else
     a = div(unsigned(diff), s) % T
end
```
to fully omit inner instability.

2. `checked_length` in L778 (and `smallints` in 799) should also be unstable. We'd better not miss them.

Reply:
That suggestion looks good to me

Reply:
But it seems somewhat at odds with how we typically compute the length (which tried to avoid assuming the type of the div result was T), if that matters to us

Reply:
That's true.
For `StepRange{T,S}`, the returned type should be `sizeof(S)` <= `sizeof(T) ? T : S` according to `length(r::OrdinalRange{T}) where {T}` in L723
But `S` seems a little bit unnessearray, as `T` should be able to represent the result.

Reply:
Yeah I was thinking the result can't be bigger than T